src/libsyntax/parse/parser.rs

```diff
@@ -1145,9 +1142,7 @@ impl<'a> Parser<'a> {
     {
         self.expect(bra)?;
         let result = self.parse_seq_to_before_end(ket, sep, f)?;
-        if self.token == *ket {
-            self.bump();
-        }
+        self.eat(ket);
```

Comment:
This should change diagnostics by adding `ket` to the expected set, but it looks like a desirable change in this case.